underdocs-renderer/src/main/kotlin/underdocs/renderer/output/html/render/section/SectionRenderer.kt

```diff
@@ -310,31 +313,45 @@ class SectionRenderer(private val linker: Linker) {
             h2("See Also")
     ).withClass("see-also")
 
-    fun renderReturnValue(visitable: Visitable, returnValue: String): Tag<*> {
-        val elements = ArrayList<DomContent>()
-
-        elements.add(h2("Return Value"))
-
-        if (visitable is Function) {
-            elements.add(p(
-                    code(referredTypeToString(visitable.returnType))
-            ).withClass("return-type"))
+    fun renderReturnValue(function: Function, returnValue: ReturnValue) = section(
+        div(
+            h2("Return Value"),
+            span(
+                singleLineTypeRenderer.render(function.returnType)
+            ).withClass("return-value-type")
+        ).withClass("return-value-heading"), div(
+            h3("Success"), renderReturnValueItemsTable(returnValue.success)
+        ),
+        iffPrimitive(returnValue.error.isNotEmpty()) {
+            div(
+                h3("Error"), renderReturnValueItemsTable(returnValue.error)
+            )
         }
-
-        elements.add(renderMarkdown(returnValue))
-
-        return section().with(elements).withClass("return-value")
+    ).withClass("return-value")
+
+    private fun renderReturnValueItemsTable(returnValueItemList: List<ReturnValueItem>): ContainerTag? {
+        return table(
+            tbody(
+                each(returnValueItemList) { child ->
+                    tr(
+                        td(
+                            h4(
```

Comment:
No need for the `h4`, instead you can place the `child.value` value directly in the `td`.

However, to be consistent with the other tables, you have to add the following CSS rule:

~~~~CSS
.return-value-table .return-value-name-cell {
    font-weight: bold;
}
~~~~

This grants a "bold" look to the `value` while

  * not adding any margin/padding as `h4` does,
  * not adding any semantic meaning (like, you know, being a heading).